bugdown: Trigger test failure for invalid Markdown include statements.

eeshangarg opened the pull request:
Fixes #10947.

@timabbott: If you take an existing webhook doc (say `zerver/webhooks/github/doc.md`) and change one of the include statements to refer to a non-existent macro, `zerver/tests/test_docs` should fail and print an informative error message. Thanks! :)

timabbott commented:
@eeshangarg cool, this is great.  I have one question about the implementation -- could `MarkdownIncludeWrapper` subclass the main `MarkdownInclude` class, instead of being a separate extension?  I haven't looked at code, but I could imagine that doing a useful deduplication...

eeshangarg commented:
@timabbott: I just updated the code. The reason I implemented a separate extension instead of subclassing was that the resulting implementation (after subclassing) wasn't necessarily simpler or functionally better in any way. In addition, subclassing meant maintaining the extension on our end, which seemed more work than just maintaining a "wrapper" that just raised an exception and fulfilled our specific purpose for the time being. :) Ideally, this should be fixed upstream! If not, we can just keep this one! Thanks!

timabbott commented:
Merged, thanks @eeshangarg!  If it's easy, can you submit a PR upstream for raising this exception?  I think it'd increase the probability that we get to stop maintaining this eventually.
